### PR TITLE
Added Translations To The Questions Index Page

### LINKED ISSUE #9040: Continued: Add Translation To The Questions Index Page
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the files linked below in the https://github.com/publiclab/plots2 repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

This is a continuation of this issue https://github.com/publiclab/plots2/issues/9033
Once this PR https://github.com/publiclab/plots2/pull/9038 has been merged you can begin working on this one. If you know how to resolve merge conflicts feel free to comment below and I can assign it to you.

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

Add translation lines for the question index page in the` config/locales/en.yml`
https://github.com/publiclab/plots2/blob/2b2bd90290f22d5a5ff709b25b1a40e11094db7e/config/locales/en.yml#L661-L664
Add the translations from `ask_question` to `liked`

```
  questions:
    index:
      explanation: "Ask a question to the Public Lab community to help you in your
        environmental exploration."
      ask_question: Ask a question here
      to_ask_question: To ask a question, please
      log_in: log in
      or: or
      sign_up: sign up
      first: first.
      recently_asked: Recently Asked
      popular: Popular
      unanswered: Unanswered
      liked: Liked
```

https://github.com/publiclab/plots2/blob/2b2bd90290f22d5a5ff709b25b1a40e11094db7e/app/views/questions/index.html.erb#L23-L25
```diff
-<h4>Ask a question here</h4>
+<h4><%=translation('questions.index.ask_question')%></h4>
<% if !current_user %>
-<p>To ask a question, please <a href="/login?return_to=/questions">log in</a> or <a href="/signup?return_to=/questions">sign up</a> first.</p>
+<p><%=translation('questions.index.to_ask_question')%><a href="/login?return_to=/questions"><%=translation('questions.index.log_in')%></a> <%=translation('questions.index.or')%> <a href="/signup?return_to=/questions"><%=translation('questions.index.sign_up')%></a><%=translation('questions.index.first')%></p>
```
https://github.com/publiclab/plots2/blob/2b2bd90290f22d5a5ff709b25b1a40e11094db7e/app/views/questions/index.html.erb#L53-L73
```diff
- <i class="fa fa-refresh"></i> <span class="d-none d-lg-inline"> Recently Asked</span>
+<i class="fa fa-refresh"></i> <span class="d-none d-lg-inline"> <%=translation('questions.index.recently_asked')%></span>
</a>
</li>
<li class="nav-item">
<a class="nav-link <% if params[:action] == "recently_commented" %> active <% end %>" href="/questions/recently_commented/">
-<i class="fa fa-check"></i> <span class="d-none d-lg-inline"> Recently Commented</span>
+<i class="fa fa-check"></i> <span class="d-none d-lg-inline"> <%=translation('questions.index.recently_commented')%></span>
</a>
</li>
<li class="nav-item">
<a class="nav-link <% if params[:action] == "popular" %> active <% end %>" href="/questions/popular/">
-<i class="fa fa-bolt"></i> <span class="d-none d-lg-inline"> Popular</span>
+<i class="fa fa-bolt"></i> <span class="d-none d-lg-inline"> <%=translation('questions.index.popular')%></span>
</a>
</li>
<li class="nav-item">
<a class="nav-link <% if params[:action] == "unanswered" %> active <% end %>" href="/questions/unanswered/">
-<i class="fa fa-question-circle"></i> <span class="d-none d-lg-inline"> Unanswered</span>
+<i class="fa fa-question-circle"></i> <span class="d-none d-lg-inline"> <%=translation('questions.index.unanswered')%></span>
</a>
</li>
<li class="nav-item">
<a class="nav-link <% if params[:action] == "liked" %> active <%end %>" href="/questions/liked/">
-<i class="fa fa-star-o"></i> <span class="d-none d-lg-inline"> Liked</span>
+<i class="fa fa-star-o"></i> <span class="d-none d-lg-inline"> <%=translation('questions.index.liked')%></span>
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by 1liale**
Fixes #9040 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/views/questions/index.html.erb [hunk 1]
```diff
@@ -20,9 +20,10 @@
 
   <div class="row">
     <div class="col-lg-6">
-      <h4>Ask a question here</h4>
+      <h4><%=translation('questions.index.ask_question')%)></h4>
       <% if !current_user %>
-        <p>To ask a question, please <a href="/login?return_to=/questions">log in</a> or <a href="/signup?return_to=/questions">sign up</a> first.</p>
+        <p><%=translation('questions.index.to_ask_question')%>, please <a href="/login?return_to=/questions"><%=translation('questions.index.log_in')%></a><%=translation('questions.index.or')%><a href="/signup?return_to=/questions"><%=translation('questions.index.sign_up')%></a>
```
Comment (RuthNjeri) [inline app/views/questions/index.html.erb:25]: Thanks for opening this up @1liale 😄 

In the yaml file, the `to_ask_question` contains `please`...   `to_ask_question: To ask a question, please` You can remove it from the line...

```suggestion
        <p><%=translation('questions.index.to_ask_question')%> <a href="/login?return_to=/questions"><%=translation('questions.index.log_in')%></a><%=translation('questions.index.or')%><a href="/signup?return_to=/questions"><%=translation('questions.index.sign_up')%></a> 
```

Comment (1liale) [reply app/views/questions/index.html.erb]: @RuthNjeri I added in the suggested change.

**Review by RuthNjeri — APPROVED** [review]
Thanks @1liale, it looks good to me 🎉

**Comment by RuthNjeri** [comment]
@1liale I have just noticed that the tests are failing...please have a look at the errors, they are related to the current changes...
https://github.com/publiclab/plots2/pull/9057/checks?check_run_id=1742921247
Screenshot 2021-01-25 at 16 39 23

Currently looking at it to share how you can solve it..

File: app/views/questions/index.html.erb [hunk 2]
```diff
@@ -20,9 +20,10 @@
 
   <div class="row">
     <div class="col-lg-6">
-      <h4>Ask a question here</h4>
+      <h4><%=translation('questions.index.ask_question')%)></h4>
```
Comment (RuthNjeri) [inline app/views/questions/index.html.erb:23]: This is causing the tests to fail
```suggestion
      <h4><%=translation('questions.index.ask_question')%></h4>
```

**Review by RuthNjeri — CHANGES_REQUESTED** [review]
This is causing the tests to fail
```suggestion
      <h4><%=translation('questions.index.ask_question')%></h4>
```

**Comment by RuthNjeri** [comment]
Hi @1liale are you able to complete work on this issue? It only needs a few changes. If not, it's no problem, we could reassign it to someone else.

**Comment by 1liale** [comment]
@RuthNjeri  yep no problem I can get it done tonight. Sorry I've been a bit absent lately. I've been really busy with co-op job searches and school work recently.

**Comment by RuthNjeri** [comment]
Ni worries @1liale, you can update me once you're done

**Comment by 1liale** [comment]
Changes have been made. Sorry I should have been more careful, there was an extra round bracket on line 23 that caused the error.

**Comment by RuthNjeri** [comment]
Hi @1liale, the changes look good so far, there seems to be a conflicting file, please rebase to ensure that the en.yml has the most recent changes as well. Let me know if you have any more questions.

**Comment by 1liale** [comment]
@RuthNjeri  Could you guide me through? I'm not completely comfortable with rebasing yet and I know that rebasing can be dangerous if not performed correctly.  Since I made changes to en.yml in two separate commits, do I only rebase those two commits? 

image

image

(these are the two commits I made to en.yml)

**Comment by RuthNjeri** [comment]
Hi @1liale, 

You can rebase the changes locally or on this PR.

1. To change it on this PR, just click the `Resolve Conflicts` button 
Screenshot 2021-02-01 at 16 13 06
2. You'll see these lines marked
Screenshot 2021-02-01 at 16 10 22
3. You should remove the text that shows the difference in the  conflicting  at lines `<<<<<main ` (line 666),`=======`(line 678), and line "686"
4. After that, you can remove any repeated lines such as` recently_commented`
5. It should look like this, also note that some of the lines from your changes don't have the double quotes. You should add them for consistency, line 677-682
Screenshot 2021-02-01 at 16 10 56
6.  Once you have made all the changes you can click the `Mark as resolved` button to commit the changes.

The other way you could rebase, is through the terminal.
1. Pull the upsteam changes to the main branch locally. It ensures that you have the most recent changes
Screenshot 2021-02-01 at 16 22 03
2. Switch to you local branch `main` Run the command `git rebase main` and it should direct you to the files that have conflicts and you can resolve them manually depending on the editor you are using.
3. You can have a look at the step by step approach to this [here](https://gist.github.com/RuthNjeri/82f3adeac98c2343a0d7e832da3993b5)

Also, when working on a new issue in the future, you should first create a new branch then push the changes here. Have a look at this [document](https://publiclab.org/notes/warren/11-22-2017/use-git-and-github-to-contribute-and-improve-public-lab-software) you can follow those steps in the future..on the site it mentions the `master` branch, assume that it means the `main` branch.

**Comment by 1liale** [comment]
@RuthNjeri  Thank you Ruth. You've been really patient with me. This has been a great learning experience for me 😄 . I rebased on this PR and fingers crossed, I really do hope that it works out now. Have a nice day.

**Review by Sagarpreet — APPROVED** [review]
Amazing team work 🎉

**Review by RuthNjeri — APPROVED** [review]
Great job @1liale this looks good to me...

@cesswairimu will merge it once reviewed :)

**Comment by RuthNjeri** [comment]
Hi @cesswairimu, this PR is ready for a merge.

**Comment by cesswairimu** [comment]
Thanks everyone..merging now

**Pull request merged**